exporter/datadogexporter/factory.go

```diff
@@ -93,7 +104,8 @@ type factory struct {
 
 	wg sync.WaitGroup // waits for agent to exit
 
-	registry *featuregate.Registry
+	registry  *featuregate.Registry
+	logsAgent logsagentpipeline.LogsAgent
```

Comment:
I don't think we need to store a reference to the pipeline within the factory struct. We use it from a single method and from lambdas created in that method, so it should be enough to store it as a local variable in that method.